src/components/tipRecords/TipRecord.vue

```diff
@@ -206,18 +210,21 @@ export default {
       return isPinned !== -1;
     },
   },
+  mounted() {
+    if (this.$route.name !== 'tip') {
+      return;
+    }
+    const { method, challenge, signature } = this.$route.query;
+    if (!this.useSdkWallet && method && challenge && signature) {
+      const cb = method === 'sendPostReport' ? this.showSuccessModalCallback : this.updatePinnedItems;
+      const params = method !== 'sendPostReport' ? [this.account] : [];
+      applyBackendChanges(method, { challenge, signature }, params, cb);
+    }
```

Comment:
if we have several components with this handler, won't `applyBackendChanges` be called multiple times?

Reply:
That's why we do a redirect to tip page - to get the tip specific methods. Similar would be for comments.

Reply:
I mean, the tip page will render several TipRecords and each of them will call `applyBackendChanges`, won't it works in this way?

Reply:
No, the tip page opens the single tip we would like to pin/report and would execute the pin or report method we are executing. Please check it out.